tests/conftest.py

```diff
@@ -1902,42 +1903,59 @@ def rgw_bucket_factory_session(request, rgw_obj_session):
 
 
 @pytest.fixture()
-def bucket_factory(request, mcg_obj):
+def bucket_factory(request, bucket_class_factory, mcg_obj):
     """
     Returns an MCG bucket factory
     """
-    return bucket_factory_fixture(request, mcg_obj)
+    return bucket_factory_fixture(request, bucket_class_factory, mcg_obj)
 
 
 @pytest.fixture(scope="session")
-def bucket_factory_session(request, mcg_obj_session):
+def bucket_factory_session(request, bucket_class_factory_session, mcg_obj_session):
     """
     Returns a session-scoped MCG bucket factory
     """
-    return bucket_factory_fixture(request, mcg_obj_session)
+    return bucket_factory_fixture(
+        request, bucket_class_factory_session, mcg_obj_session
+    )
 
 
-def bucket_factory_fixture(request, mcg_obj=None, rgw_obj=None):
+def bucket_factory_fixture(
+    request, bucket_class_factory=None, mcg_obj=None, rgw_obj=None
+):
     """
     Create a bucket factory. Calling this fixture creates a new bucket(s).
     For a custom amount, provide the 'amount' parameter.
 
     Args:
+        bucket_class_factory: creates a new Bucket Class
         mcg_obj (MCG): An MCG object containing the MCG S3 connection
             credentials
         rgw_obj (RGW): An RGW object
 
     """
     created_buckets = []
 
-    def _create_buckets(amount=1, interface="S3", verify_health=True, *args, **kwargs):
+    def _create_buckets(
+        amount=1,
+        interface="S3",
+        verify_health=True,
+        bucketclass: typing.Union[None, str, dict] = None,
+        *args,
+        **kwargs,
+    ):
         """
         Creates and deletes all buckets that were created as part of the test
 
         Args:
             amount (int): The amount of buckets to create
             interface (str): The interface to use for creation of buckets.
                 S3 | OC | CLI | NAMESPACE
+            verify_Health (bool): Whether to verify the created bucket's health
+                post-creation
+            bucketclass (None|str|dict): An existing bucketclass name to use,
+                or a dictionary describing a new bucketclass to be created.
```

Comment:
I would prefer to have only bucket name string as parameter. It seems overly complicated to me to create new bucket class as part of this factory if dict provided.

Reply:
All test cases but one actually rely on the flow in which a new bucketclass is created.
The aim is for it to either be `None` (and then the bucket will use the default bucketclass), or `dict` (and then a bucketclass will be created).
Removing the option to create a new bucketclass as part of creating a new bucket will force us to refactor all of the parametrized tests to be two-liners instead of one-liners as they are now; Since we'll always have to create the bucketclass, and then pass its name/object to the factory anyway.

The only test that needs to use an existing bucketclass is `test_multi_region`, which was written before the current infrastructure was in place. The test needs access to the backingstores that are created for the bucket, but those are not externalized at the moment. The only way to access them is by creating a bucketclass separately, and then creating a bucket that utilizes it.

The plan is to change the design so that the ObjectBucket object will point to a BucketClass object, which will then have a list of its backingstores. Then, the option for `str` will be removed - and the factory will either use the default bucketclass, or create a new one.

Reply:
what is required in order to accept only dict here? if it's possible to remove the str option, then this arg can be set to None as default and accept only dict

Reply:
Changed the code to accept `dict` and `None`